File: 1402. Reducing Dishes/1402. Reducing Dishes.js
var maxSatisfaction = function(satisfaction) {
    satisfaction.sort((a,b)=>a-b);
    let l=satisfaction.length;
    let ans=0;
    let sum=0;
    let temp=0;
    for(let i=l-1;i>=0;i--){
        temp+=satisfaction[i];
        sum+=temp;
        ans=Math.max(sum,ans);
        if(ans<0){return 0;}
    }
    return ans;
};

File: 1402. Reducing Dishes/1402. Reducing Dishes.go
func maxSatisfaction(satisfaction []int) int {
    sort.Ints(satisfaction)
    sum:=0
    temp:=0
    for i:=len(satisfaction)-1;i>=0;i-- {
        if satisfaction[i] + sum + temp > sum {
            temp+=satisfaction[i]
            sum+=temp
        }
    }
    return sum
}
